R/plot_carp.R

```diff
@@ -514,3 +420,497 @@ carp_heatmaply <- function(x,
+            y = ~V2)
+      }
+
+      path_static %>%
+        add_text(data = plot_frame_init,
+                 x = ~V1,
+                 y = ~V2,
+                 text = ~ObsLabel,
+                 #size = label_size,
+                 inherit = FALSE) %>%
+        hide_legend() %>%
+        style(text=mytext, hoverinfo = "text", traces = 1) %>%
+        style(hoverinfo = "none", traces = c(2:(length(mytext)+2)))
+    }
+
+  }
+  else{
+    plot_frame_animation <- bind_rows(lapply(percent.seq, function(pct){
+      ## Make a list of things to plot at each "percent" and then combine
+      plot_frame_full %>% filter(.data$GammaPercent <= pct) %>%
+        mutate(Regularization = pct)
+    }))
+
+    path_dynamic<-plot_ly()
+
+    for (i in 1:length(plot_frame_init$ObsLabel)){
+      path_dynamic <- path_dynamic %>%
+        add_paths(
+          data = plot_frame_animation[plot_frame_animation$Obs==i,],
+          color = I("red"),
+          ids = ~Obs,
+          x = ~V1,
+          y = ~V2,
+          frame = ~Regularization*100)
+    }
+
+    path_dynamic %>%
+      add_markers(
+        data = plot_frame_interactive,
+        ids = ~Obs,
+        x = ~V1,
+        y = ~V2,
+        color = I("red"),
+        size = 2.5,
+        frame = ~Regularization*100) %>%
+      add_markers(
+        data = plot_frame_init,
+        ids = ~Obs,
+        x = ~V1,
+        y = ~V2,
+        color = I("black"),
+        size = 2,
+        inherit = FALSE) %>%
+      add_text(data = plot_frame_init,
+               x = ~V1,
+               y = ~V2,
+               text = ~ObsLabel,
+               #size = label_size,
+               inherit = FALSE) %>%
+      hide_legend() %>%
+      style(text=mytext, hoverinfo = "text", traces = c(length(mytext)+2)) %>%
+      style(hoverinfo = "none", traces = c(1:(length(mytext)+1))) %>%
+      animation_slider(currentvalue = list(prefix = "Regularization: ",suffix = "%")) #%>%
+  }
+}
+
+#' @noRd
+#' @importFrom rlang .data
+#' @importFrom dplyr filter select summarize pull desc
+#' @importFrom stats as.dendrogram is.leaf
+#' @importFrom dendextend get_nodes_xy
+#' @importFrom plotly add_segments add_markers add_text plot_ly hide_legend highlight animation_slider
+#' @importFrom RColorBrewer brewer.pal
+#' @importFrom tibble as_tibble
+carp_dendro_plotly <- function(x,
+                             dynamic = FALSE,
+                             percent,
+                             percent.seq,
+                             k,
+                             show_clusters = (n_args == 1L),
+                             slider_y = -0.3,
+                             ...
+){
+
+  has_percent <- !missing(percent)
+  has_k       <- !missing(k)
+  n_args      <- has_percent + has_k
+
+  if(n_args > 1L){
+    crv_error("At most one of ", sQuote("percent"), " and ", sQuote("k"), " may be supplied.")
+  }
+
+  d <- as.dendrogram(x)
+  # get x/y locations of every node in the tree
+  m <- dendextend::get_nodes_xy(d)
```

Comment:
This shouldn't need to be qualified with `dendextend` if the NAMESPACE is working properly .